src/Sylius/Bundle/ApiBundle/Changer/PaymentMethodChanger.php

```diff
@@ -51,6 +53,7 @@ public function changePaymentMethod(
                 'Can not change payment method for this payment',
             );
             $payment->setMethod($paymentMethod);
+            $this->paymentRequestCanceller->cancelPaymentRequests($paymentId, $paymentMethodCode);
```

Comment:
This method doesn't check if **source** and **target** `paymentMethodCode` are the same, which means it will cancel existing payment method request for valid **target** `paymentMethodCode`.

What do you think adding changing payment method from same to same  code to the scenarios?

Reply:
The check is actually inside of The `cancelPaymentRequest()` method

Reply:
Ah, I see now.

What about the case, when order is being cancelled fully and I want to cancel current payment requests with currently selected payment method?

I think this needs to be refactored, only `changer` should be aware of this case of source/target method change and `canceller` should just cancel (Do one job and don't do support logic for `changer`)